---
parent: Classes
title: nmdc:OrthologyGroup
grand_parent: Browse the NMDC Schema
layout: default
---

# Type: OrthologyGroup


A set of genes or gene products in which all members are orthologous

URI: [nmdc:OrthologyGroup](https://microbiomedata/meta/OrthologyGroup)


---

![img](http://yuml.me/diagram/nofunky;dir:TB/class/[FunctionalAnnotationTerm]%5E-[OrthologyGroup%7Chas_raw_value(i):string%20%3F],[OntologyClass],[FunctionalAnnotationTerm],[Activity])

---


## Identifier prefixes

 * KEGG.ORTHOLOGY
 * EGGNOG
 * PFAM
 * TIGRFAM
 * SUPFAM
 * PANTHER.FAMILY

## Parents

 *  is_a: [FunctionalAnnotationTerm](FunctionalAnnotationTerm.md) - Abstract grouping class for any term/descriptor that can be applied to a functional unit of a genome (protein, ncRNA, complex).

## Attributes


### Inherited from attribute value:

 * [has raw value](has_raw_value.md)  <sub>OPT</sub>
    * Description: The value that was specified for an annotation in raw form, i.e. a string. E.g. "2 cm" or "2-4 cm"
    * range: [String](types/String.md)

### Inherited from controlled term value:

 * [term](term.md)  <sub>OPT</sub>
    * Description: pointer to an ontology class
    * range: [OntologyClass](OntologyClass.md)

### Inherited from functional annotation:

 * [functional annotation➞was generated by](functional_annotation_was_generated_by.md)  <sub>OPT</sub>
    * Description: provenance for the annotation. Note to be consistent with the rest of the NMDC schema we use the PROV annotation model, rather than GPAD
    * range: [Activity](Activity.md)
 * [functional annotation➞subject](functional_annotation_subject.md)  <sub>OPT</sub>
    * range: [GeneProduct](GeneProduct.md)
 * [functional annotation➞has function](functional_annotation_has_function.md)  <sub>OPT</sub>
    * range: [FunctionalAnnotationTerm](FunctionalAnnotationTerm.md)

## Other properties

|  |  |  |
| --- | --- | --- |
| **Exact Mappings:** | | biolink:GeneFamily |

